### PR TITLE
fix: respect cache render options

### PR DESCRIPTION
**Pull request by mastodon0**
I tried to disable the cache for a single renderFile call and discovered this bug. 
The passed in cache option would always be ignored.

### CODE REVIEW

File: src/liquid-options.ts [hunk 1]
```diff
@@ -89,11 +89,13 @@ export function normalize (options?: LiquidOptions): NormalizedOptions {
...
-  else if (typeof options.cache === 'object') cache = options.cache
-  else cache = options.cache ? new LRU<Template[]>(1024) : undefined
-  options.cache = cache
+  if (options.hasOwnProperty('cache')) {
+    let cache: Cache<Template[]> | undefined
+    if (typeof options.cache === 'number') cache = options.cache > 0 ? new LRU(options.cache) : undefined
+    else if (typeof options.cache === 'object') cache = options.cache
+    else cache = options.cache ? new LRU<Template[]>(1024) : undefined
+    options.cache = cache
+  }
```
Comment (harttle) [inline src/liquid-options.ts:98]: Is this change necessary?

File: test/integration/liquid/cache.ts [hunk 2]
```diff
@@ -175,4 +190,19 @@ describe('LiquidOptions#cache', function () {
       expect(y).to.equal('foo')
     })
   })
+  it('should respect passed in cache=false option', async function () {
```
Comment (harttle) [inline test/integration/liquid/cache.ts:193]: seems in the wrong level, should be inside describe renderFileSync

File: src/liquid-options.ts (hunk 1 above)
Comment (mastodon0) [reply src/liquid-options.ts:98]: Yes, the change in [liquid.ts#78](https://github.com/harttle/liquidjs/blob/90b863104e721a6041cf9d0f75de99d0d2767dda/src/liquid.ts#L78) would make couple of other tests crash. The options-object would always have the cache-key set to undefined which would overwrite the base cache.
I added two tests (sync/async) to demonstrate and test this problem

File: test/integration/liquid/cache.ts (hunk 2 above)
Comment (mastodon0) [reply test/integration/liquid/cache.ts]: Sorry about that, fixed

**Comment by harttle** [comment]
@all-contributors please add mastodon0 for code.
